Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1332,6 +1332,21 @@ namespace OpenInfraPlatform {
 									loops.push_back(loop);
 								}
 								
+								std::vector<carve::geom::vector<3>> firstPointloop = loops[0];
+								std::vector<carve::geom::vector<3>> secondPointLoop = loops[1];
+
+								carve::geom::vector<3> firstPoint = firstPointloop[0];
+								carve::geom::vector<3> secondPoint = firstPointloop[1];
+								carve::geom::vector<3> thirdPoint = firstPointloop[2];
+
+								carve::geom::vector<3> firstToSecond = secondPoint - firstPoint;
+								carve::geom::vector<3> firstToThird = thirdPoint - firstPoint;
+
+								carve::geom::vector<3> normalOfPlane = carve::geom::cross(firstToSecond, firstToThird);
+								normalOfPlane.normalized();

```

Comment:
I'd introduce a function that takes a vector of points and returns the normal.

Please also have a look at the code in the new function `correctWinding` - there, plane normals are already calculated. So preferably, there would be one function that both call. (I vaguely remember that such a function should already exist in GeomUtils - did you have a look?)

Reply:
Resolved: 4db3b19